### DISCUSSION TITLE
[SOLVED] Directus 9.0.0 rc-93 custom interface problem

### DISCUSSION THREAD

**Discussion by korneliusz-w** [directus/directus#8188] [Archive]
Hi 
I just upgraded directus from rc-88 to rc-93 and my custom interface stop working.
When I run directus it loads extensions two times. There is no error in CMS, i can select it in field definition but I can't use it in collection.
```
09:28:39 Loaded extensions: select-image, 404
09:28:39 PUBLIC_URL should be a full URL
09:28:40 PostGIS isn't installed. Geometry type support will be limited.
09:28:40 Loaded extensions: select-image, 404
09:28:40 Server started at http://localhost:8055
```

**Comment by rijkvanzanten** [comment]
There's not enough info here to really be able to help out much..

What does your interface look like? Are you adhering to the structure as per the docs? Are you using the extensions-sdk? How are you bundling your custom interface?

**Comment by AustinPhillipTaylor** [comment]
Can't be sure from the amount of information given, however, be sure that you're bundling with the extensions-sdk, it was renamed to extensions, with an "s", in rc.92. The backend replaces the extensions-sdk dependency with the compiled version (along with other shared/global dependencies) when the server is started. If your interface was built with the extension-sdk, but the server is looking for the extensions-sdk, it will create issues.

**Comment by korneliusz-w** [comment]
Sorry, for lack of information

EDIT: Something was changed between rc-88 and rc-93 because my requests for files is not working now.  Template is working when I put some text like <p>My custom interface</p> 

EDIT 2: I found. Instead of inject: ['system'], now should be inject: ['api'],  but the info of this is only in custom module example https://docs.directus.io/guides/modules/
Now it is working

My package.json 

```
{
  "name": "select-image",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@directus/extensions-sdk": "^9.0.0-rc.93"
  },
  "directus:extension": {
    "type": "interface",
    "path": "dist/index.js",
    "source": "src/index.js",
    "host": "^9.0.0-rc.93",
    "hidden": false
  }  
}
```

My interface

```
import InterfaceComponent from './interface.vue';

export default {
	id: 'select-image',
	name: 'Select image',
	description: 'Wybierz zdjęcie',
	icon: 'box',
	component: InterfaceComponent,
	types: ['string'],
};
```

```
<template>
	<select :value="value" v-if="collections" @change="handleChange($event.target.value)">
		<option value="">---Wybierz opcję---</option>
		<option v-for="col in collections" v-bind:value="col.value">{{ col.text }}</option>
	</select>
</template>

<script>
export default {
	data() {
		return {			
			collections: null
		};
	},	
	emits: ['input'],
	props: {
		value: String,
	},	
	methods: {
		handleChange(value) {			
			this.$emit('input', event.target.value);
		},	
	},
	inject: ['system'],          /// should be inject: ['api']
	mounted() {
		// Get a list of all available collections to use with this module
		this.system.api.get('/files').then((res) => {                /// should be this.api.get
			this.collections = res.data.data.map(item => {
				return {
					text: item.filename_download,
					value: item.id
				}
			}).sort((a, b) => {
				if (a.text < b.text) {
					return -1;
				} else if (a.text > b.text) {
					return 1;
				} else {
					return 0;
				}
			});
		});
	},	
};
</script>
```

Build
```
$ npx directus-extension build
- Building Directus extension...
√ Done
```

Than I move index.js from /dist to extensions/interface/select-image and run

```
D:\directus>npx directus start
10:05:17 Loaded extensions: select-image
10:05:17 PUBLIC_URL should be a full URL
10:05:17 PostGIS isn't installed. Geometry type support will be limited.
10:05:17 Loaded extensions: select-image
10:05:17 Server started at http://localhost:8055
```

As you can see the extendions are loading two times. In CMS I can select it from field interface but when I try to use it in collection it is not availabe - blank field

**Reply by joselcvarela** [reply]
I think it logged twice because of some change that trigger a reload.
In fact this is logged in `initializeExtensions` and this function is only called once https://github.com/directus/directus/search?q=initializeExtensions.

**Reply by rijkvanzanten** [reply]
@nickrum any other changes required with the interface updates?

**Reply by nickrum** [reply]
I don't think so. The interface works on my end using the `api` inject.

Extensions are loaded twice because of #7675. This will be fixed by #7891.
